lib/bambora/rest/batch_payment_file_upload_client.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module Bambora
+  module Rest
+    class BatchPaymentFileUploadClient < Bambora::Rest::Client
+      CONTENT_DISPOSITION = 'form-data'
+      FILE_CONTENT_TYPE = 'text/plain'
+      FILE_TYPE_HEADER = { 'FileType' => 'STD' }
+      JSON_CONTENT_TYPE = 'application/json'
+
+      # @param args[:file_contents] [String] CSV file contents
+      # @param args[:options] [Hash] Request Parameters as documented by Bambora.
+      # @param args[:api_key] [String]
+      def post(args = {})
+        init_payload(args[:file_contents], args[:options])
+        parse_response_body(
+          super(
+            path: args[:path],
+            body: @payload.body,
+            headers: build_headers(api_key: args[:api_key]),
+          ),
+        ).to_h
+      end
+
+      private
+
+      def init_payload(file_contents, options)
+        return @payload if @payload
+
+        @payload = Bambora::Adapters::MultipartMixedRequest.new(
```

Comment:
is it memoization in a different way?

Reply:
Yeah, Rubocop gave me an error because it expects memoized methods to have the same name as the instance variable. But it seemed more semantic to me to name this method `init_` since the instance variable is used elsewhere and it doesn't make sense to use a method that accepts args as an accessor.

Reply:
Yeah, Rubocop gave me an error because it expects memoized methods to have the same name as the instance variable. But it seemed more semantic to me to name this method `init_` since the instance variable is used elsewhere and it doesn't make sense to use a method that accepts args as an accessor.

Reply:
Yeah, Rubocop gave me an error because it expects memoized methods to have the same name as the instance variable. But it seemed more semantic to me to name this method `init_` since the instance variable is used elsewhere and it doesn't make sense to use a method that accepts args as an accessor.

Reply:
Yeah, Rubocop gave me an error because it expects memoized methods to have the same name as the instance variable. But it seemed more semantic to me to name this method `init_` since the instance variable is used elsewhere and it doesn't make sense to use a method that accepts args as an accessor.